storage/memory/storage.go

```diff
@@ -202,6 +203,20 @@ func (r ReferenceStorage) SetReference(ref *plumbing.Reference) error {
 	return nil
 }
 
+func (r ReferenceStorage) CheckAndSetReference(ref, old *plumbing.Reference) error {
```

Comment:
This function is not covered by the tests.

Reply:
Ah, I didn't notice that the remote tests don't use filesystem storage.  I'll add a test.